src/EVM/SymExec.hs

```diff
@@ -646,19 +646,19 @@ equivalenceCheck solvers bytecodeA bytecodeB opts calldata' = do
         -- if we can statically determine that the end states differ, then we
         -- ask the solver to find us inputs that satisfy both sets of branch
         -- conditions
-        PBool True  -> Just . Set.fromList $ aProps <> bProps
+        PBool True  -> Just . Set.fromList $ extractProps aEnd <> extractProps bEnd
```

Comment:
So before we would never really take into account the props inside expressions for equivalence checking? Does this make a difference semantically?

Reply:
oh true, it was probably a soundness bug. didn't even notice when I was making the change...